src/main/java/uk/gov/pay/directdebit/mandate/model/MandateBankStatementReferenceArgumentFactory.java

```diff
@@ -0,0 +1,21 @@
+package uk.gov.pay.directdebit.mandate.model;
+
+import org.jdbi.v3.core.argument.AbstractArgumentFactory;
+import org.jdbi.v3.core.argument.Argument;
+import org.jdbi.v3.core.config.ConfigRegistry;
+
+import java.sql.Types;
+
+public class MandateBankStatementReferenceArgumentFactory extends AbstractArgumentFactory<MandateBankStatementReference> {
```

Comment:
Wondering if there's a way to create an argument factory that would work for all objects that are just wrapping a String since it's seems its always just a null safe `.toString()`; similar to the `ToStringSerializer` so we wouldn't have to keep creating these for each specific type.

Reply:
Yeah, I’ve been wondering about that too: we have several of these argument factories now and they’re all nearly identical. Maybe it’s doable if all the types have a common supertype.